src/inspect_ai/model/_model_output.py

```diff
@@ -9,6 +9,24 @@
 from ._chat_message import ChatMessageAssistant
 
 
+class PerfectCacheUsage(BaseModel):
+    """
+    Running totals **under the assumption of a perfect KV-cache**.
+    Identical semantics to IdealizedModelUsage but recorded per-task so that
+    code that runs later in the same asyncio Task can retrieve it.
+    """
+
+    input_tokens: int = Field(default=0)  # ALL prompt tokens
+    unique_input_tokens: int = Field(default=0)  # cache misses
+    output_tokens: int = Field(default=0)  # completions
+
+    def __iadd__(self, other: "PerfectCacheUsage") -> "PerfectCacheUsage":
+        self.input_tokens += other.input_tokens
+        self.unique_input_tokens += other.unique_input_tokens
+        self.output_tokens += other.output_tokens
+        return self
+
```

Comment:
Per my earlier comment, seems like we might just want to use a `ModelUsage` object instead.